src/uart.rs

```diff
@@ -458,14 +593,32 @@ impl<'d> UartDriver<'d> {
         rts: Option<impl Peripheral<P = impl OutputPin> + 'd>,
         config: &config::Config,
     ) -> Result<Self, EspError> {
-        new_common(uart, Some(tx), Some(rx), cts, rts, config)?;
+        let mut q_handle_raw = ptr::null_mut();
+        let q_handle = if config.queue_size > 0 {
+            Some(&mut q_handle_raw)
+        } else {
+            None
+        };
+        new_common(uart, Some(tx), Some(rx), cts, rts, config, q_handle)?;
 
         Ok(Self {
             port: UART::port() as _,
+            queue: q_handle_raw,
             _p: PhantomData,
         })
     }
 
+    /// Retrieves the event queue for this UART. Returns `None` if
+    /// the config specified 0 for `queue_size`.
+    pub fn event_queue(&self) -> Option<Queue<uart_event_t>> {
```

Comment:
More importantly, I don't feel comfortable exposing the raw `uart_event_t` as the primary API. Can we use a native Rust enum here?

Reply:
Unless we provide some special magic to map all the values returned by the queue or implement a wrapper around `Queue` specifically for UART events, I don't see a way.

Reply:
How about the following:
- `#[repr(transparent)]` on a new struct, `UartEvent(uart_event_t)`
- The UART queue would be returning `UartEvent`
- A method: `UartEvent::payload(&self) -> UartEventPayload` or `UartEvent::payload(self) -> UartEventPayload`
- `UartEventPayload` can then be our proper Rust enum, with `From<uart_event_t>` implementation
- `UartEvent` and `UartEventPayload` are both `Copy, Clone, Eq, PartialEq`; `UartEventPayload` is also `Debug`  

...because `UartEvent` is `#[repr(transparent)]` you can safely transmute between it and `uart_event_t`.

Reply:
... and obviously `event_queue` would be returning `&Queue<UartEvent>` using transmute.